src/main/java/com/firebolt/jdbc/type/BaseType.java

```diff
@@ -124,9 +139,9 @@ private static boolean isNan(String value) {
 		return "nan".equalsIgnoreCase(value) || "+nan".equalsIgnoreCase(value) || "-nan".equalsIgnoreCase(value);
 	}
 
-	private static void validateObjectNotNull(String value) {
+	private static void validateObjectNotNull(String value) throws SQLException {
 		if (value == null) {
-			throw new IllegalArgumentException("The value cannot be null");
+			throw new FireboltException("The value cannot be null");
 		}
 	}
```

Comment:
Let's try improving the error message here since by using FireboltException we're removing some of the information. Perhaps adding what operation went wrong would be better?

Reply:
reverted. In fact this cannot happen because value cannot be null here.

Reply:
Although I disagree that replacing IllegalArgumentException by FireboltException (i.e. SQLException) looses any information.

Reply:
I think the information that the argument caused this error is helpful (at least partially). FireboltException class is generic and can fit many errors, including some backend ones, while IllegalArgumentException points at the argument provided. I'm not saying one is preferable over another, but that it helps narrowing down where the issue is coming from.
In any case, it's a minor point.